conda_devenv/gen_scripts.py

```diff
@@ -0,0 +1,378 @@
+import shlex
+from pathlib import Path
+from textwrap import dedent
+
+from typing import Callable, Dict, Iterable, List, Union, NamedTuple
+
+
+Value = Union[str, List[str]]
+Environment = Dict[str, Value]
+
+
+class ScriptRenderer(NamedTuple):
+    """Renders a script.
+
+    Should be provided a preamble with needed boilerplate, a function capable
+    of generating a script body based on the environment, and an optional
+    epilogue.
+    """
+
+    preamble: str
+    generate_body: Callable[[Environment], str]
+    epilogue: str = ""
+
+    def render(self, environment: Environment) -> str:
+        """Render the script.
+
+        :param environment: The environment to base the script on.
+        :return: A full script.
+        """
+        return "\n".join(
+            s
+            for s in (
+                f"{self.preamble}",
```

Comment:
How about adding in all preambles a comment like "This file was generated by conda-devenv"?

Reply:
Done!